state-chain/pallets/cf-elections/src/electoral_systems/solana_vault_swap_accounts.rs

```diff
@@ -165,8 +165,10 @@ impl<
 				let mut known_accounts = election_access.properties()?;
 				election_access.delete();
 				known_accounts.witnessed_open_accounts.extend(consensus.new_accounts.iter().map(
-					|(account, swap_details)| {
-						Hook::initiate_vault_swap(swap_details.clone());
+					|(account, maybe_swap_details)| {
+						if let Some(swap_details) = maybe_swap_details.as_ref() {
+							Hook::initiate_vault_swap(swap_details.clone());
+						}
```

Comment:
If someone submits invalid data they will never find out - we log it in the engines but maybe we should also notify somehow (an event?).

Reply:
That sounds reasonable but at this stage we could only emit the event account address, which will be closed afterwards. We will only be able to fetch that on-chain information for a limited period of time.
I guess the most useful part would be for external parties to have a way to identify if their swap has been detected and has failed or not. Having said that, for other chains (at least EVM) I believe the data doesn't even make it to the SC if it fails decoding.
Therefore I'm not sure this adds more information that just monitoring the engine for decoding errors.